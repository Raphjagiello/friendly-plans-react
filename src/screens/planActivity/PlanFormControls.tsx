import { FormikProps } from 'formik';
import React, { SFC } from 'react';
import { StyleSheet, View } from 'react-native';

import PlayButton, { TextInput } from 'components';
import { i18n } from 'locale';
import { Plan } from 'models';
import { dimensions, getElevation, palette } from 'styles';
import { Icon, StyledText } from '../../components';
import { PlanFormData } from './PlanForm';
import { ShuffleButton } from './ShuffleButton';

interface Props extends FormikProps<PlanFormData> {
  plan: Plan;
}

export const PlanFormControls: SFC<Props> = ({ values, handleChange, isSubmitting, submitForm, plan }) => (
  <View style={styles.container}>
    <View style={styles.inputContainer}>
      <Icon name="emoticon" size={24} color={palette.textInputPlaceholder} />
      {isSubmitting ? (
        <StyledText style={styles.styledText}>{values.planInput}</StyledText>
      ) : (
        <TextInput
          style={styles.textInput}
          placeholder={i18n.t('planActivity:planNamePlaceholder')}
          value={values.planInput}
          onChangeText={handleChange('planInput')}
          onEndEditing={submitForm}
        />
      )}
    </View>
    <View style={styles.buttonContainer}>
      <ShuffleButton disabled={!isSubmitting} />
      <PlayButton plan={plan} disabled={!isSubmitting} size={50} />
    </View>
  </View>
);

const styles = StyleSheet.create({
  container: {
    height: 56,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    paddingLeft: dimensions.spacingLarge,
    paddingRight: dimensions.spacingBig,
    backgroundColor: palette.background,
    ...getElevation(5),
  },
  inputContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    maxWidth: '40%',
  },
  buttonContainer: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  styledText: {
    color: palette.textBlack,
    marginLeft: dimensions.spacingSmall,
  },
  textInput: {
    marginLeft: dimensions.spacingSmall,
  },
});
