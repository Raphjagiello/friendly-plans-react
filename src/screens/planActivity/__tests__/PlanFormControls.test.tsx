import { shallow } from 'enzyme';
import React from 'react';
import { PlanFormControls } from '../PlanFormControls';

describe('PlanFormControls', () => {
  const props = {
    handleChange: jest.fn(),
    values: {
      planInput: 'Music',
    },
    submitForm: jest.fn(),
    isSubmitting: false,
  } as any;

  it('renders correctly', () => {
    const planInput = shallow(<PlanFormControls {...props} />);

    expect(planInput).toMatchSnapshot();
  });
});
