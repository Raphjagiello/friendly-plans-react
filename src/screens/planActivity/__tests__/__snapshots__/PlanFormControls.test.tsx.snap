// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlanFormControls renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "elevation": 5,
      "flexDirection": "row",
      "height": 56,
      "justifyContent": "space-between",
      "paddingLeft": 32,
      "paddingRight": 24,
      "shadowColor": "rgba(0, 0, 0, 0.7)",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.22000000000000003,
      "shadowRadius": 3.5,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "maxWidth": "40%",
      }
    }
  >
    <Icon
      color="#C8CBFA"
      name="emoticon"
      size={24}
    />
    <Component
      disabled={false}
      onEndEditing={[MockFunction]}
      value="Music"
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Component
      disabled={false}
    />
    <Component
      disabled={true}
      size={50}
    />
  </View>
</View>
`;
